lib/huginn_agent.rb

```diff
@@ -1,17 +1,45 @@
 require 'huginn_agent/version'
 
 class HuginnAgent
-  def self.register(*paths)
-    @paths ||= []
-    paths.each do |path|
-      @paths << path
+  class << self
+    attr_accessor :branch, :remote
+
+    def load_tasks(options = {})
+      @branch = options[:branch] || 'master'
+      @remote = options[:remote] || 'https://github.com/cantino/huginn.git'
+      Rake.add_rakelib File.join(File.expand_path('../', __FILE__), 'tasks')
+    end
+
+    def load(*paths)
+      paths.each do |path|
+        load_paths << path
+      end
+    end
+
+    def register(*paths)
```

Comment:
Agreed that it's nice how little code change was needed.  Agents definitely should stay subclasses of Agent for now, I think.  All of our AR code assumes this.  It'd be okay to add an (optional?) registration call made at the class level when an Agent is defined, though.  Or just a class method that sets a class variable indicating the group or something @TildeWill?